local_planner/src/nodes/waypoint_generator.cpp

```diff
@@ -360,16 +370,19 @@ void WaypointGenerator::adaptSpeed() {
       wrapPolar(p_pol_fcu);
       speed_ *= scaleToFOV(fov_fcu_frame_, p_pol_fcu);
     }
-
-    // Scale the pose_to_wp by the speed
-    Eigen::Vector3f pose_to_wp = output_.goto_position - position_;
-    if (pose_to_wp.norm() > 0.1f) pose_to_wp.normalize();
-    pose_to_wp *= std::min(speed_, goal_dist);
-
     heading_at_goal_rad_ = NAN;
-    output_.adapted_goto_position = position_ + pose_to_wp;
   }
 
+  speed_ = lp_filter_constant * speed_ + (1.f - lp_filter_constant) * last_speed;
```

Comment:
Be aware that the real time constant of your filter will change if dt changes. 
A safer implementation is:
```
alpha = dt / (time_constant + dt)
out = alpha * in + (1 - alpha) * out_prev
```
if the `time constant >> dt`, then you can simplify the formula by  `alpha = dt * time_constant_inverse` and give `time_constant_inverse` as a `static constexpr`